src/users/users.controller.ts

```diff
@@ -7,36 +7,42 @@ import {
   HttpStatus,
   Param,
   Post,
+  Res,
 } from '@nestjs/common';
 import { CreateUserDto, LoginUserDto } from '../dto/users-dto';
 import { UsersService } from './users.service';
+import { Response } from 'express';
 
 @Controller('users')
 export class UsersController {
   constructor(private readonly usersService: UsersService) {}
 
   @Post('/signup')
-  createUser(@Body() createUserDto: CreateUserDto): object {
-    const data = this.usersService.createUser(createUserDto);
-    return data;
+  async createUser(
+    @Body() createUserDto: CreateUserDto,
+    @Res() res: Response,
+  ): Promise<object> {
+    const data = await this.usersService.createUser(createUserDto);
+    const { token } = data;
+    res.header('Authorization', `Bearer ${token}`);
+    console.log(data);
+    return res.status(HttpStatus.OK).json(data);
```

Comment:
Don't have to return the whole `data`, jut the user info of the data

Reply:
emm...I think better to return all data include token , because sometime frontend will use response token and save token to localstorge.